File: app/src/main/java/com/example/androidtvapptutorial/Movie.java
package com.example.androidtvapptutorial;

import android.util.Log;

import java.net.URI;
import java.net.URISyntaxException;

public class Movie {

    private static final String TAG = Movie.class.getSimpleName();

    static final long serialVersionUID = 727566175075960653L;
    private long id;
    private String title;
    private String studio;
    private String cardImageUrl;

    public Movie() {

    }

    /*public String getCardImageUrl() {
        return cardImageUrl;
    }

    public void setCardImageUrl(String cardImageUrl) {
        this.cardImageUrl = cardImageUrl;
    }*/

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getStudio() {
        return studio;
    }

    public void setStudio(String studio) {
        this.studio = studio;
    }

    public String getCardImageUrl() {
        return cardImageUrl;
    }

    public void setCardImageUrl(String cardImageUrl) {
        this.cardImageUrl = cardImageUrl;
    }

    @Override
    public String toString() {
        return "Movie{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", cardImageUrl='" + cardImageUrl + '\'' +
                '}';
    }

    /*public URI getCardImageURI() {
        try {
            return new URI(getCardImageUrl());
        } catch (URISyntaxException e) {
            Log.e("Error",e.toString());
            return null;
        }
    }*/
}

File: app/src/main/java/com/example/androidtvapptutorial/Movie.kt
package com.example.androidtvapptutorial

class Movie {

    private val TAG = Movie::class.java.getSimpleName()

    val serialVersionUID = 727566175075960653L

    private var id : Long? = null
    private var title: String? = null
    private var studio: String? = null

    constructor()

    fun getId() : Long{
        return id?:0
    }

    fun setId(id : Long){
        this.id = id
    }

    fun  getTitle() : String{
        return title?:""
    }

    fun setTitle(title : String){
        this.title = title
    }

    fun getStudio(): String {
        return studio?:""
    }

    fun setStudio(studio: String) {
        this.studio = studio
    }

    override fun toString(): String {
        return "Movie{" +
                "id=" + id +
                ", title='" + title + '\''.toString() +
                '}'.toString()
    }
}